backend/main/linting.py

```diff
@@ -1,28 +1,89 @@
 import ast
-
 from pyflakes import checker
-from pyflakes.messages import UnusedVariable
+from pyflakes.messages import UnusedImport, UnusedVariable, IsLiteral, RedefinedWhileUnused, ImportShadowedByLoopVar, \
+    ImportStarUsed, ImportStarUsage, ImportStarNotPermitted, DuplicateArgument, MultiValueRepeatedKeyLiteral
 
 MESSAGES = {
+    UnusedImport: """
+**Unused import `{0}`**
+You imported a module `{0}` but never used it. Did you forget to use it?
+Maybe you used the wrong import in its place? If you don't need it, just remove it entirely.
+    """,
     UnusedVariable: """
 **Unused variable `{0}`**
-
 You defined a variable `{0}` but never used it. Did you forget to use it?
 Maybe you used the wrong variable in its place? If you don't need it, just remove it entirely.
-    """
+    """,
+    IsLiteral: """
+**Is literal**
+You used the is/is not statement for comparison. You should have rather used the == / != statements,
+which are used to compare constant literals (str, bytes, int, float, tuple)- referring to equality. 
+The is/is not statement checks if objects refer to the same instance (address in memory).
+    """,
+
+    RedefinedWhileUnused: """
+**Redefined `{0}` while unused on line `{1}`**
+This error occurs when a function, class or method is redefined.
+The function `{0}` has not been called however it has been redefined on line `{1}`
+Make sure that all functions have different names if they are different. Also remember to call the function.
+    """,
+    ImportShadowedByLoopVar: """
+**Import `{0}` from line '{1}` shadowed by loop variable**
+The name of the loop variable `{0}` should be changed in line`{1}` as it redefines the `{0}` module.
+Choose a different loop variable to avoid this error.
+""",
+    ImportStarUsed: """
+
+**Import made using * **
+This * import is used to import everything from a designated module under the current 
+module, allowing the use of various objects from the imported module- without having to prefix them with the module's 
+name. Refrain from using this type of import statement and rather explicitly import a few statements that you may 
+require instead.  
+""",
+    ImportStarUsage: """
+
+**Import made using * **
+This * import is used to import everything from a designated module under the current 
+module, allowing the use of various objects from the imported module- without having to prefix them with the module's 
+name. Refrain from using this type of import statement and rather explicitly import a few statements that you may 
+require instead.   
+""",
+    ImportStarNotPermitted: """
+
+**Import made using * **
+This * import is used to import everything from a designated module under the current 
+module, allowing the use of various objects from the imported module- without having to prefix them with the module's 
+name. Refrain from using this type of import statement and rather explicitly import a few statements that you may 
+require instead.
+""",
```

Comment:
Try to emit the same warning only once. If it's not possible, explain why and use a variable instead of copying the whole message.

Reply:
Can You please elaborate on what you mean by emitting the same warning once. Does that have to do with the fact that some commands may result in 2 warnings being thrown?

Reply:
`for x in lint(code_import_star):` prints the same message twice. See if we can have fewer entries in this dict and still get a warning whenever they import star.

Reply:
Ok taking out the `ImportStarUsage `entry. The `ImportStarUsed` entry should cover all instances of a star being imported.